ctapipe/analysis/tests/test_sensitivity.py

```diff
@@ -0,0 +1,219 @@
+                                          make_mock_event_rate, crab_source_rate)
+
+
+def test_check_min_N():
+    N = [2, 2]
+
+    min_N = 10
+
+    scale = check_min_N(N, min_N=min_N)
+
+    assert sum(N) == min_N
+
+
+def test_check_background_contamination():
+    N = [2, 2]
+
+    max_prot_ratio = .1
+
+    scale = check_background_contamination(N, max_prot_ratio=max_prot_ratio)
+
+    assert N[1] / sum(N) == max_prot_ratio
+
+
+def test_SensitivityPointSource_effective_area():
+
+    # areas used in the event generator
+    gen_area_g = np.pi*(1000*u.m)**2
+    gen_area_p = np.pi*(2000*u.m)**2
+
+    # energy list of "selected" events
+    energy_sel_gamma  = np.logspace(2, 6, 200, False) * u.TeV
+    energy_sel_elect  = np.logspace(2, 6, 200, False) * u.TeV
+    energy_sel_proton = np.logspace(2, 6, 400, False) * u.TeV
+
+    # energy list of "generated" events
+    energy_sim_gamma  = np.logspace(2, 6, 400, False) * u.TeV
+    energy_sim_elect  = np.logspace(2, 6, 400, False) * u.TeV
+    energy_sim_proton = np.logspace(2, 6, 800, False) * u.TeV
+
+    # binning for the energy histograms
+    energy_edges = np.logspace(2, 6, 41) * u.TeV
+
+    # energy histogram for the generated events
+    energy_sim_hist_gamma = np.histogram(energy_sim_gamma,
+                                         bins=energy_edges)[0]
+    energy_sim_hist_elect = np.histogram(energy_sim_elect,
+                                         bins=energy_edges)[0]
+    energy_sim_hist_proton = np.histogram(energy_sim_proton,
+                                          bins=energy_edges)[0]
+
+    # constructer gets fed with energy and angular offset lists and desired energy binning
+    Sens = SensitivityPointSource(
+                    mc_energies={'g': energy_sel_gamma,
+                                 'p': energy_sel_proton,
+                                 'e': energy_sel_elect},
+                    energy_bin_edges={"g": energy_edges,
+                                      "p": energy_edges,
+                                      'e': energy_edges})
+
+    # effective areas
+    Sens.get_effective_areas(
+                    generator_energy_hists={'g': energy_sim_hist_gamma,
+                                            'p': energy_sim_hist_proton,
+                                            'e': energy_sim_hist_elect},
+                    generator_areas={'g': gen_area_g,
+                                     'p': gen_area_p,
+                                     'e': gen_area_g})
+
+    # midway result are the effective areas
+    eff_a = Sens.effective_areas
+    eff_area_g, eff_area_p, eff_area_e = eff_a['g'], eff_a['p'], eff_a['e']
+
+    # "selected" events are half of the "generated" events
+    # so effective areas should be half of generator areas, too
+    np.testing.assert_allclose(eff_area_g.value, gen_area_g.value/2)
+    np.testing.assert_allclose(eff_area_p.value, gen_area_p.value/2)
+
+
+def test_SensitivityPointSource_sensitivity_MC():
```

Comment:
@cdeil I added a specific test for the sensitivity. Is this what you had in mind?